File: app/src/main/java/hegde/mahesh/avre/interpreter/LanguageInterpreter.java
package hegde.mahesh.avre.interpreter;

import java.io.OutputStream;

public interface LanguageInterpreter {
    void setVariable(String variable, Object value) throws SnippetEvalException;
    void setOutputStream(OutputStream out);
    void setErrorStream(OutputStream err);
    Object eval(String snippet) throws SnippetEvalException;
    @SuppressWarnings("unused")
    String[] getCompletions(String partialInput, int position) throws SnippetEvalException;
    void reset();
}

File: app/src/main/java/hegde/mahesh/avre/interpreter/LanguageInterpreter.kt
package hegde.mahesh.avre.interpreter

import java.io.OutputStream

interface LanguageInterpreter {
    @Throws(SnippetEvalException::class)
    fun setVariable(variable: String, value: Any?)
    fun setOutputStream(out: OutputStream)
    fun setErrorStream(err: OutputStream)

    @Throws(SnippetEvalException::class)
    fun eval(snippet: String): Any?

    @Throws(SnippetEvalException::class)
    fun getCompletions(partialInput: String, position: Int): Array<String>
    fun reset()
}